java/vector/src/test/java/org/apache/arrow/vector/complex/writer/TestComplexWriter.java

```diff
@@ -801,7 +801,7 @@ public void timeStampMicroWriters() throws Exception {
   public void timeStampNanoWriters() throws Exception {
     // test values
     final long expectedNanos = 981173106123456789L;
-    final LocalDateTime expectedNanoDateTime = new LocalDateTime(2001, 2, 3, 4, 5, 6, 123);
+    final LocalDateTime expectedNanoDateTime = LocalDateTime.of(2001, 2, 3, 4, 5, 6, 123456789);
```

Comment:
I don't think this was quite right before, there were missing nanoseconds in LocalDateTime. Maybe they were being truncated in the result?